src/mapping_navigation/scripts/navigation.py

```diff
@@ -1,34 +1,59 @@
 #!/usr/bin/env python3
 
 import rospy
+import os
 from std_msgs.msg import Float64, String
-from mapping_navigation.msg import PathData
 from geometry_msgs.msg import PoseStamped
+from pure_pursuit import *
+from std_msgs.msg import Float64
+from simple_path import *
 
 
 class Navigation:
 
     """
-    This does mapping
+    This does navigation
     """
 
-    def __init__(self):
+    def __init__(self, path: List[Point], look_ahead_distance: float, look_forward_gain: float, wheel_base: float):
         self.steering_pub = rospy.Publisher('steering', Float64, queue_size=10)
-        self.curve_pub = rospy.Publisher('curve_data', String, queue_size=10)
-        self.path_data = rospy.Publisher('pathData', PathData, queue_size=10)
+        self.path = path
+        self.target_index = 0
+        self.navigator = PurePursuit(look_ahead_distance, look_forward_gain, path)
+        self.state = State(wheel_base)
 
     def listener(self):
-        rospy.init_node('talker', anonymous=True)
+        rospy.init_node('navigation', anonymous=True)
         rospy.Subscriber('airsimPose', PoseStamped, self.handle_gps_data)
-        rospy.spin()
-
-    def handle_gps_data(self, data):
-        self.steering_pub.publish(1)
-        self.curve_pub.publish('curve_data')
-        self.path_data.publish(PathData())
-        print('Handles gps data')
-
+        rospy.Subscriber("sensor/speed", Float64, self.handle_speed_data)        
+        rate = rospy.Rate(30)
+        # Once to get initial starting index
+        self.target_index, _ = self.navigator.search_target_index(self.state)
```

Comment:
What's the point of the underscore variable? It's not used later

Reply:
It's cause its not used later, I have seen that notation used in the past to say ignore this, will replace this though so that the underscore is not present.

Reply:
For what it's worth, I've also seen that notation to indicate that you need to bind a result to something but not use it. The underscore, at least in other languages, allows a tuple to be unpacked but to show only part of a tuple will be used